src/main/java/io/github/thebusybiscuit/slimefun4/core/services/CustomItemDataService.java

```diff
@@ -45,7 +45,14 @@ public void setItemData(ItemMeta im, String id) {
     }
 
     public Optional<String> getItemData(ItemStack item) {
-        return getItemData(item.getItemMeta());
+        if (!item.hasItemMeta()) {
+            return Optional.empty();
+        }
+        ItemMeta meta = item.getItemMeta();
+        if (meta == null) {
+            return Optional.empty();
+        }
```

Comment:
This is redundant, ItemStack#getItemMeta() will **never** return null, except for ItemStackWrappers but we already handled these in the previous condition. The null-check I suggested was about the ItemStack, not the meta.
`if (item == null || !item.hasItemMeta())`

Reply:
why would the item be null

Reply:
If I do `.getItemData(null)` the ItemStack will be null.

Reply:
it did not have a null check for however long this class has existed before, that seems out of the scope of this PR to change

Reply:
Something being broken for a long time doesn't mean you shouldn't fix it.
And it is very well in scope of this PR since your change broke this method. It's only logical to fully fix this then, rather than doing a half-hearted job and fixing it only halfway through, otherwise the issue will only be delayed to a more distant point in time.